### ISSUE TITLE
[BUG] formatting issue with enum default values in help text

### ISSUE THREAD

**Issue by asieira** [tiangolo/typer#290]
### Describe the bug

Under certain circumstances, enum Option defaults are displayed at `EnumClass.value` instead of just `value`. I was able to reproduce it when a list of enums is expected, but there might be more scenarios.

### To Reproduce

Steps to reproduce the behavior with a minimum self-contained file.

Replace each part with your own scenario:

* Create a file `main.py` with:

```Python
from enum import Enum
from typing import List

import typer

app = typer.Typer()

class NeuralNetwork(str, Enum):
    SIMPLE = "simple"
    CONV = "conv"
    LSTM = "lstm"

@app.command()
def main(
        network: List[NeuralNetwork] = typer.Option([NeuralNetwork.SIMPLE], case_sensitive=False)
):
    typer.echo(f"Training neural network of type: {network.value}")


if __name__ == "__main__":
    app()
```

* Call it with:

```bash
python main.py --help
```

* It outputs:

```
Usage: main.py [OPTIONS]

Options:
  --network [simple|conv|lstm]    [default: NeuralNetwork.SIMPLE]
  --install-completion [bash|zsh|fish|powershell|pwsh]
                                  Install completion for the specified shell.
  --show-completion [bash|zsh|fish|powershell|pwsh]
                                  Show completion for the specified shell, to
                                  copy it or customize the installation.

  --help                          Show this message and exit.
```

* But I expected it to output:

```
Usage: main.py [OPTIONS]

Options:
  --network [simple|conv|lstm]    [default: simple]
  --install-completion [bash|zsh|fish|powershell|pwsh]
                                  Install completion for the specified shell.
  --show-completion [bash|zsh|fish|powershell|pwsh]
                                  Show completion for the specified shell, to
                                  copy it or customize the installation.

  --help                          Show this message and exit.
```

### Expected behavior

I expected the enum string value to be displayed, not including the class name which is just confusing for the CLI user.

### Environment

* OS: macOS 11.4 (Intel)
* Typer Version [e.g. 0.3.0], get it with: 0.3.2
* Python: 3.9.5

**Comment by asieira** [comment]
Seems to be related to https://github.com/tiangolo/typer/blob/master/typer/core.py#L71

**Comment by captainCapitalism** [comment]
I checked this out and you are right and wrong. 

This is wrong:
The code references you are showing is related to Arguments and it is not executing with Option like in your example. In reality, the formatting happens here:
https://github.com/pallets/click/blob/7.1.2/src/click/core.py#L1887
or in 8.0 click:
https://github.com/pallets/click/blob/main/src/click/core.py#L2733

the logic in both of these and the Typer code you pointed to is basically the same so it is a detail. You can confirm it when using debugger.

The right is:
the exact same thing will happen when using Argument.

It is something that I don't understand but looks like it is related to `format` method on string.

You can check out the behaviour of enum:
```python
from enum import Enum


class NeuralNetwork(str, Enum):
    SIMPLE = "simple"
    CONV = "conv"
    LSTM = "lstm"


print("First set:")
print("print: ", NeuralNetwork.SIMPLE)
print("single; this happens with single value: {}".format(NeuralNetwork.SIMPLE))
print("list: {}".format([NeuralNetwork.SIMPLE]))
print("casted: {}".format(str(NeuralNetwork.SIMPLE)))
print("casted list; this is coded: {}".format([str(NeuralNetwork.SIMPLE)]))
print("list of values: {}".format([NeuralNetwork.SIMPLE.value]))
```
The result is:
```bash
First set:
print:  NeuralNetwork.SIMPLE
single; this happens with single value: simple
list: [<NeuralNetwork.SIMPLE: 'simple'>]
casted: NeuralNetwork.SIMPLE
casted list; this is coded: ['NeuralNetwork.SIMPLE']
list of values: ['simple']
```

You can fix this from your side by overwriting `__str__` method on your enum, the default is:
```python
def __str__(self):
    return "%s.%s" % (self.__class__.__name__, self._name_)
```
So you could do something like that:
``` python

from enum import Enum

class NeuralNetwork2(str, Enum):
    SIMPLE = "simple"
    CONV = "conv"
    LSTM = "lstm"

    def __str__(self):
        return self.value


print("Second set:")
print("print: ", NeuralNetwork2.SIMPLE)
print("single; this happens with single value: {}".format(NeuralNetwork2.SIMPLE))
print("list: {}".format([NeuralNetwork2.SIMPLE]))
print("casted: {}".format(str(NeuralNetwork2.SIMPLE)))
print("casted list; this is coded: {}".format([str(NeuralNetwork2.SIMPLE)]))
print("list of values: {}".format([NeuralNetwork2.SIMPLE.value]))
```
Here result is:
```bash
Second set:
print:  simple
single; this happens with single value: simple
list: [<NeuralNetwork2.SIMPLE: 'simple'>]
casted: simple
casted list; this is coded: ['simple']
list of values: ['simple']
```
The modified enum is fixing the help for me.

**Comment by asieira** [comment]
Seems like Typer should be mapping Enum instances to [click.Choice](https://click.palletsprojects.com/en/8.0.x/options/#choice-opts) based on the enum values, no?

**Comment by jalaziz** [comment]
Just ran into this. Here's a super simple reproduction without a list:

```
#!/usr/bin/env python3

from enum import Enum

import typer

app = typer.Typer()


class Choice(str, Enum):
    foo = "foo"
    bar = "bar"


@app.command()
def select_choice(choice: Choice = Choice.foo):
    print(choice)


if __name__ == "__main__":
    app()
```

**Comment by asieira** [comment]
I submitted https://github.com/tiangolo/typer/pull/473 to help address this, not sure how to help further to get this issue addressed. 🤔

**Comment by tiangolo** [comment]
Thanks for the fix @asieira! :rocket: 

This will be available in the next few hours in Typer 0.9.2 :rocket:
